pkgs/development/tools/misc/gdb/wrapper.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, lib, makeWrapper, gdb-unwrapped, safePaths }:
+
+let
+  gdb = gdb-unwrapped;
+in
+  stdenv.mkDerivation {
+    name = gdb.name;
+    buildInputs = [ makeWrapper ];
+    propagatedBuildInputs = [ gdb ];
+    propagatedUserEnvPkgs = [ gdb ];
```

Comment:
Please avoid `propagatedUserEnvPkgs`. It's not needed either.

Reply:
If removed the man-page for gdb is not available. Is there an alternative?

Reply:
The wrapper could become part of the package itself.

Reply:
So I should apply my changes in `gdb/default.nix` instead?

Reply:
For sake of simplicity I would actually prefer gdb not being a script.
Instead I would suggest to compile stdenv.cc.cc.lib as default safe path into gcc: https://sourceware.org/gdb/onlinedocs/gdb/Auto_002dloading-safe-path.html using `--with-auto-load-safe-path`. Since libgcc is a dependency of gdb this would not increase closure size. There might be other packages that could be of interests such as `go` in safe-path. However I would rather put this into a nixos/homemanager module since not everybody uses those.

Reply:
Personally I just have the whole `/nix/store` in my safe path, since I trust the programs that I am debugging in general.

Reply:
> `--with-auto-load-safe-path`

Ah, so there is a configure-time option for this; I'll try it out.

> I just have the whole `/nix/store`

Does this not require you to trust your whole store? I don't, so I have ad-hoc white-listed the libstd++ plugin every update before this patch.

Reply:
@tmplt right, however I personally don't consider this a problem, since I don't attach GDB to random processes.

Reply:
Latest commit adds a configuration flag and scaps the wrapper.